### PR TITLE
Tenant foundation: every row always has a tenant (always-a-tenant)

### PR DESCRIPTION
**Pull request by robmillersoftware**
## Description

**Tenant foundation — "every row always has a tenant" (Wave D fast-follow, PR A of 2).** Makes tenant ownership a total invariant so that enabling multi-tenancy later can never orphan pre-existing data. This is the data-model foundation the ownership *enforcement* (PR B — the #42/#43/#54/#55/#73 checks) sits on.

Motivation: the ownership checks only work if projects/subscriptions/keys reliably carry a tenant. Today they're nullable, so a single-tenant → multi-tenant flip would strand all existing rows. Industry norm (GitLab Organizations, Rails `acts_as_tenant`, django-tenants) is NOT NULL + a permanent default tenant + backfill — never null. Contex already had `DEFAULT_TENANT_ID` + `ensure_default_tenant`; this wires it through.

Design doc: `docs/superpowers/plans/2026-09-07-tenant-foundation-always-a-tenant.md`

## Type of Change

- [x] Breaking change (schema migration; behavior only differs at scale/on tenant deletion)
- [x] New feature (tenant-ownership invariant)

## Related Issues

Relates to #42, #43, #54, #55, #73 (the ownership-enforcement fast-follows that build on this in PR B).

## Changes Made

- **Migration 007** (`alembic/versions/007_*`): ensures the `'default'` tenant exists (fresh-DB-safe), backfills `NULL tenant_id → 'default'` on `subscriptions` / `api_keys` / `service_accounts` / `events`, backfills a `tenant_projects` row for every project found in events/embeddings/subscriptions, sets `server_default='default'` + `NOT NULL` on those four columns, adds the missing `subscriptions → tenants` FK, and switches `api_keys`/`service_accounts` FKs to `ondelete=RESTRICT` (events keeps CASCADE). The `events` NOT NULL uses the zero-downtime `NOT VALID → VALIDATE → SET NOT NULL → DROP` pattern to avoid a long `ACCESS EXCLUSIVE` scan on a large event-sourced table.
- `SubscriptionService.create()` now defaults `None → 'default'` so it can never write an explicit NULL.
- Fixed a pre-existing bug: `tenant_routes.get_tenant_manager` constructed `TenantManager(redis)` instead of `TenantManager(db)`.
- Decoupled the invariant from the `MULTI_TENANT_ENABLED` flag: the column is *always* populated; the flag only governs enforcement (PR B).
- Auxiliary tables (`audit_logs`, `webhook_endpoints`, `agent_registrations`, `APIKeyRole.tenant_id`) are intentionally **out of scope** — same defect, deferred to a sweep if we keep those subsystems.

## Testing

- [x] All existing tests pass
- [x] New tests added for new functionality
- [x] Tested edge cases

Full `tests/` suite: **433 passed, 0 failed**. New: `tests/test_migration_tenant_backfill.py` (server-default lands 'default', NOT NULL raises IntegrityError, `create()` defaults). The always-a-tenant FKs required the test harness to guarantee the `'default'` tenant exists per test — `tests/conftest.py` now preserves it across cleanups (this was the source of an 11-test blast radius the full suite caught). (~6 `sdk/python/tests` failures are pre-existing/unrelated and untouched.)

## Checklist

- [x] Self-review performed (each commit per-task reviewed; migration reviewed against fresh-DB safety, ordering, FK/ondelete, reversibility)
- [x] Documentation updated (plan committed)
- [x] No new warnings
- [x] Tests added proving the invariant
- [x] Existing tests pass locally

## Additional Notes

**Follow-up (PR B):** the ownership-enforcement checks (#42/#43/#54/#55/#73, `/sandbox/subscribe` scope, removing the dead `require_admin_permission`) build on this foundation and will come as a separate PR rebased on top.

**Operational note:** on tenant deletion, `api_keys`/`service_accounts` are now `RESTRICT` (was `SET NULL`) — deleting a tenant that still owns keys/service-accounts will be refused rather than silently orphaning them.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/core/db_models.py [hunk 1]
```diff
@@ -464,7 +464,9 @@ class Subscription(Base):
 
     subscription_id: Mapped[str] = mapped_column(String(255), primary_key=True)
     project_id: Mapped[str] = mapped_column(String(255), nullable=False)
-    tenant_id: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
+    tenant_id: Mapped[str] = mapped_column(
```
Comment (robmillersoftware) [inline src/core/db_models.py]: Since we're doing this 5 times, do you think this is worth handling via inheritance? That way if we add any more tables that carry a tenant_id, we'll have a standardized way of doing that.

File: alembic/versions/007_tenant_not_null_backfill.py [hunk 2]
```diff
@@ -79,9 +79,21 @@ def upgrade() -> None:
         op.alter_column(table, 'tenant_id', server_default='default')
 
     # 5. NOT NULL on the four columns (backfill guarantees no NULLs remain).
```
Comment (robmillersoftware) [inline alembic/versions/007_tenant_not_null_backfill.py]: 3 columns?

File: alembic/versions/007_tenant_not_null_backfill.py [hunk 3]
```diff
@@ -79,9 +79,21 @@ def upgrade() -> None:
...
-    for table in ('subscriptions', 'api_keys', 'service_accounts', 'events'):
+    #    Small tables: plain SET NOT NULL. events is event-sourced and can be
+    #    large in prod, so apply NOT NULL via a NOT VALID CHECK + VALIDATE to
+    #    avoid a full-table scan under ACCESS EXCLUSIVE; PG12+ then skips the
+    #    scan on SET NOT NULL because the validated CHECK already proves it.
```
Comment (robmillersoftware) [inline alembic/versions/007_tenant_not_null_backfill.py]: That's too many words.

File: alembic/versions/007_tenant_not_null_backfill.py [hunk 4]
```diff
@@ -129,6 +141,8 @@ def downgrade() -> None:
     # Drop the subscriptions FK.
     op.drop_constraint('fk_subscriptions_tenant', 'subscriptions', type_='foreignkey')
 
-    # Re-add nullable, drop server_default on the four columns.
+    # Re-add nullable, drop server_default on the four columns. Also drop the
+    # transient events CHECK in case a downgrade runs after a partial upgrade.
```
Comment (robmillersoftware) [inline alembic/versions/007_tenant_not_null_backfill.py]: I don't think we need this in-depth of a comment here.

File: src/core/subscriptions.py [hunk 5]
```diff
@@ -22,6 +23,8 @@ def __init__(self, db, matcher, redis) -> None:
     async def create(
         self, project_id, needs, tenant_id=None, scope=None, subscription_id=None, top_k=None, threshold=None
     ) -> str:
+        if tenant_id is None:
+            tenant_id = DEFAULT_TENANT_ID
```
Comment (robmillersoftware) [inline src/core/subscriptions.py]: Why not just set the default in the parameter list?

File: tests/conftest.py [hunk 6]
```diff
@@ -98,6 +99,33 @@ async def db() -> AsyncGenerator[DatabaseManager, None]:
...
+                text(
+                    """
+                    INSERT INTO tenant_usage (tenant_id, projects_count, agents_count, api_keys_count,
+                                             events_this_month, storage_used_mb, last_updated)
+                    VALUES (:tid, 0, 0, 0, 0, 0.0, now())
+                    ON CONFLICT (tenant_id) DO NOTHING
+                    """
+                ),
+                {"tid": DEFAULT_TENANT_ID},
+            )
```
Comment (robmillersoftware) [inline tests/conftest.py]: Can we not use an ORM for this?

File: src/core/db_models.py (hunk 1 above)
Comment (robmillersoftware) [reply src/core/db_models.py]: Extracted a `TenantScopedMixin` (tenant_id column + server_default + NOT NULL + FK to tenants), applied to the 4 tenant-FK tables (subscriptions/api_keys/service_accounts/events). events overrides ondelete to CASCADE via `_tenant_ondelete`; the composite-PK tenant tables (TenantProject/TenantUsage) are left alone. (7f86b1d)

File: alembic/versions/007_tenant_not_null_backfill.py (hunk 2 above)
Comment (robmillersoftware) [reply alembic/versions/007_tenant_not_null_backfill.py]: Right — that loop is the 3 small tables; events is handled separately below. Reworded the comment to say so. (fc56b87)

File: alembic/versions/007_tenant_not_null_backfill.py (hunk 3 above)
Comment (robmillersoftware) [reply alembic/versions/007_tenant_not_null_backfill.py]: Trimmed to one line. (fc56b87)

File: alembic/versions/007_tenant_not_null_backfill.py (hunk 4 above)
Comment (robmillersoftware) [reply alembic/versions/007_tenant_not_null_backfill.py]: Trimmed the downgrade comment. (fc56b87)

File: src/core/subscriptions.py (hunk 5 above)
Comment (robmillersoftware) [reply src/core/subscriptions.py]: Done — `tenant_id: str = DEFAULT_TENANT_ID` in the signature, dropped the None-normalize block. (ddf00dd)

File: tests/conftest.py (hunk 6 above)
Comment (robmillersoftware) [reply tests/conftest.py]: Switched to `ensure_default_tenant(...)` (goes through the ORM via TenantManager; it also creates the tenant_usage row). Left the bulk-DELETE cleanup as raw SQL since it is bulk truncation. (9e2e347)
